Patch origin: in-house
Patch status: Solaris-specific; not suitable for upstream

Redirects build process to use local archives instead of downloading
from external webs.

Forces older version of bnd as new version 7.0.0 is built for JDK 17
("class file has wrong version 61.0, should be 52.0" issue).
Note that bnd is used for creating OSGi metadata in JAR files only
(it's not part of distribution).

--- apache-tomcat-9.0.83-src/build.properties.default
+++ apache-tomcat-9.0.83-src/build.properties.default
@@ -122,6 +122,12 @@ base-maven.loc=https://repo.maven.apache
 # Mirror, was used when there were problems with the main SF downloads site
 # base-sf.loc=https://sunet.dl.sourceforge.net
 
+# In Userland we really don't want Tomcat to download its build
+# dependencies itself but instead we will do it ourself in advance.
+# Therefore those dependencies which are really required need to be
+# redirected to local file system.
+userland-files.loc=file://${base.path}/../../
+
 # ----- Webservices - JAX RPC -----
 jaxrpc-lib.version=1.1-rc4
 jaxrpc-lib.checksum.enabled=true
@@ -129,7 +135,7 @@ jaxrpc-lib.checksum.algorithm=MD5|SHA-1
 jaxrpc-lib.checksum.value=4bebba22a4cdb9f68e16c45129770333|fe9371d33dc3e1646d4d13bde19614283eb998b1
 jaxrpc-lib.home=${base.path}/jaxrpc-${jaxrpc-lib.version}
 jaxrpc-lib.jar=${jaxrpc-lib.home}/geronimo-spec-jaxrpc-${jaxrpc-lib.version}.jar
-jaxrpc-lib.loc=${base-maven.loc}/geronimo-spec/geronimo-spec-jaxrpc/${jaxrpc-lib.version}/geronimo-spec-jaxrpc-${jaxrpc-lib.version}.jar
+jaxrpc-lib.loc=${userland-files.loc}/geronimo-spec-jaxrpc-${jaxrpc-lib.version}.jar
 
 # ----- Webservices - WSDL4J -----
 wsdl4j-lib.version=1.6.3
@@ -138,7 +144,7 @@ wsdl4j-lib.checksum.algorithm=MD5|SHA-1
 wsdl4j-lib.checksum.value=cfc28d89625c5e88589aec7a9aee0208|6d106a6845a3d3477a1560008479312888e94f2f
 wsdl4j-lib.home=${base.path}/wsdl4j-${wsdl4j-lib.version}
 wsdl4j-lib.jar=${wsdl4j-lib.home}/wsdl4j-${wsdl4j-lib.version}.jar
-wsdl4j-lib.loc=${base-maven.loc}/wsdl4j/wsdl4j/${wsdl4j-lib.version}/wsdl4j-${wsdl4j-lib.version}.jar
+wsdl4j-lib.loc=${userland-files.loc}/wsdl4j-${wsdl4j-lib.version}.jar
 
 # ----- Eclipse JDT, version 4.7 or later -----#
 # See https://cwiki.apache.org/confluence/display/TOMCAT/Managing+Tomcat%27s+Dependency+on+the+Eclipse+JDT+Core+Batch+Compiler
@@ -152,13 +158,13 @@ # https://download.eclipse.org/eclipse/d
 #
 jdt.version=4.20
 jdt.release=R-4.20-202106111600
-jdt.checksum.enabled=true
+jdt.checksum.enabled=false
 jdt.checksum.algorithm=SHA-512
 jdt.checksum.value=ab441acf5551a7dc81c353eaccb3b3df9e89a48987294d19e39acdb83a5b640fcdff7414cee29f5b96eaa8826647f1d5323e185018fe33a64c402d69c73c9158
 jdt.home=${base.path}/ecj-${jdt.version}
 jdt.jar=${jdt.home}/ecj-${jdt.version}.jar
 # The download will be moved to the archive area eventually. We are taking care of that in advance.
-jdt.loc.1=http://archive.eclipse.org/eclipse/downloads/drops4/${jdt.release}/ecj-${jdt.version}.jar
+jdt.loc.1=${userland-files.loc}/ecj-${jdt.version}.jar
 jdt.loc.2=http://download.eclipse.org/eclipse/downloads/drops4/${jdt.release}/ecj-${jdt.version}.jar
 
 # ----- Tomcat native library -----
@@ -304,16 +310,17 @@ spotbugs.loc=${base-maven.loc}/com/githu
 
 # ----- bnd, version 6.3.0 or later  -----
 # ----- provides OSGI metadata for JARs       -----
-bnd.version=7.0.0
+bnd.version=6.4.1
 
 # checksums for biz.aQute.bnd-6.4.1.jar
 bnd.checksum.enabled=true
 bnd.checksum.algorithm=MD5|SHA-1
-bnd.checksum.value=654776477ed942fc53f581fec66e253a|9937f6b7528628964a4ab8e50ba6b964d0310bce
+bnd.checksum.value=78a21ab586850832a78342b8d0a99f64|91f28d145a1284f270a3d2cdc886863460c89833
 
 bnd.home=${base.path}/bnd-${bnd.version}
 bnd.jar=${bnd.home}/biz.aQute.bnd-${bnd.version}.jar
 bnd.loc=${base-maven.loc}/biz/aQute/bnd/biz.aQute.bnd/${bnd.version}/biz.aQute.bnd-${bnd.version}.jar
+bnd.loc=${userland-files.loc}/biz.aQute.bnd-${bnd.version}.jar
 
 # ----- JSign, version 4.1 or later -----
 jsign.version=5.0
@@ -348,3 +355,11 @@ derby-shared.jar=${derby.home}/derby-sha
 derby-shared.loc=${base-maven.loc}/org/apache/derby/derbyshared/${derby.version}/derbyshared-${derby.version}.jar
 derby-tools.jar=${derby.home}/derby-tools-${derby.version}.jar
 derby-tools.loc=${base-maven.loc}/org/apache/derby/derbytools/${derby.version}/derbytools-${derby.version}.jar
+
+# allow Tomcat unit test pass even with JDK 8
+# https://bz.apache.org/bugzilla/show_bug.cgi?id=66064
+opens.javalang=-Dnop
+opens.javaio=-Dnop
+opens.sunrmi=-Dnop
+opens.javautil=-Dnop
+opens.javautilconcurrent=-Dnop
